### PR TITLE
Remove Context and ContextKind

### LINKED ISSUE #60421: Remove rustc_mir::borrow_check::{Context, ContextKind}
`ContextKind` is completely unused in the borrow checker. It should be removed and any uses of `Context` should be replaced with `rust::mir::Location`.

### PR DESCRIPTION
**Pull request by JohnTitor**
Fixes #60421

r? @matthewjasper

### CODE REVIEW

File: src/librustc_mir/borrow_check/mod.rs [hunk 1]
```diff
@@ -2193,7 +2190,7 @@ impl<'cx, 'gcx, 'tcx> MirBorrowckCtxt<'cx, 'gcx, 'tcx> {
                                     // `*const` raw pointers are not mutable
                                     hir::MutImmutable => Err(place),
                                     // `*mut` raw pointers are always mutable, regardless of
-                                    // context. The users have to check by themselves.
+                                    // location. The users have to check by themselves.
```
Comment (matthewjasper) [inline src/librustc_mir/borrow_check/mod.rs]: Should still be "context" (being used as a common English word)

File: src/librustc_mir/borrow_check/mod.rs [hunk 2]
```diff
@@ -173,7 +173,7 @@ fn do_mir_borrowck<'a, 'gcx, 'tcx>(
     // Replace all regions with fresh inference variables. This
     // requires first making our own copy of the MIR. This copy will
     // be modified (in place) to contain non-lexical lifetimes. It
-    // will have a lifetime tied to the inference context.
+    // will have a lifetime tied to the inference location.
```
Comment (matthewjasper) [inline src/librustc_mir/borrow_check/mod.rs]: This is referring to `InferCtxt` and shouldn't be changed.

File: src/librustc_mir/borrow_check/mod.rs [hunk 3]
```diff
@@ -511,7 +512,7 @@ pub struct MirBorrowckCtxt<'cx, 'gcx: 'tcx, 'tcx: 'cx> {
     /// If the function we're checking is a closure, then we'll need to report back the list of
     /// mutable upvars that have been used. This field keeps track of them.
     used_mut_upvars: SmallVec<[Field; 8]>,
-    /// Non-lexical region inference context, if NLL is enabled. This
+    /// Non-lexical region inference location, if NLL is enabled. This
```
Comment (matthewjasper) [inline src/librustc_mir/borrow_check/mod.rs]: Should be still be "context" (referring to RegionInferenceContext)

**Review by matthewjasper — COMMENTED** [review]
There are a few places that have been changed that shouldn't be. Otherwise this looks good.

File: src/librustc_mir/borrow_check/mod.rs (hunk 1 above)
Comment (JohnTitor) [reply src/librustc_mir/borrow_check/mod.rs]: Thanks, I reverted wording.

**Pull request merged**